apps/sms-client/src/app/components/calendar/calendar.component.ts

```diff
@@ -1,21 +1,106 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { CommonModule } from '@angular/common';
 
+type GridDates = Date | null;
+
 @Component({
   selector: 'sms-calendar',
   standalone: true,
-  imports: [
-    CommonModule
-  ],
+  imports: [CommonModule],
   templateUrl: './calendar.component.html',
   styleUrl: './calendar.component.css',
 })
-export class CalendarComponent {
-  public viewDate: Date = new Date();
+export class CalendarComponent implements OnInit {
+  public currentDate: Date = new Date();
+  public selectedYear = this.currentDate.getFullYear();
+  public selectedMonth = this.currentDate.getMonth();
   public holidays: Date[] = [new Date('2024-01-01'), new Date('2024-12-25')];
+  public grid!: GridDates[][];
 
   public dayClicked(day: Date): void {
     console.log('Clicked on: ', day);
   }
+  public month = 'current month';
+  public days = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'];
+  public months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  ngOnInit(): void {
+    this.grid = this.generateCalendarGrid(
+      this.selectedYear,
+      this.selectedMonth,
+    );
+  }
+
+  generateCalendarGrid(year: number, month: number): GridDates[][] {
+    const startDate = new Date(year, month, 1);
+    const endDate = new Date(year, month + 1, 0);
+    const numDays = endDate.getDate();
+    const startDayOfWeek = startDate.getDay();
+
+    const grid: GridDates[][] = [[]];
+    let currentWeek = 0;
+    for (let i = 1; i <= startDayOfWeek; i++) {
+      grid[0].push(null);
+    }
+
+    for (let i = 1; i <= numDays; i++) {
+      const currentDate = new Date(year, month, i);
+      if (currentDate.getDay() === 0 && i !== 1) {
+        currentWeek++;
+        grid.push([]);
+      }
+      grid[currentWeek].push(currentDate);
+    }
+    return grid;
+  }
+
+  prevMonth() {
```

Comment:
run the lint to make sure the rules are applied because, you always need to have a method access modifier and return type for functions